sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1431,7 +1432,8 @@ pi_result _pi_queue::executeCommandList(pi_command_list_ptr_t CommandList,
     // event that anybody waiting for any event in the batch will
     // really be using.
     //
-    if (EventsScope == LastCommandInBatchHostVisible) {
+    if (EventsScope == LastCommandInBatchHostVisible &&
+        !CommandList->second.EventList.empty()) {
```

Comment:
Why is this extra check? Just an optimization?

Reply:
I see explanation in conversation just below, and it makes sense but please add a source comment too.

Reply:
I see explanation in conversation just below, and it makes sense but please add a source comment too.